### PR TITLE
Remove hateoas

### LINKED ISSUE #111: Cache HATEOAS returned links
## Describe your idea for a new feature
When a session is generated, create a cache for HATEOAS links, so they don't get refetched with each requests

Clear the cache with each new session. (This stuff should be done with the `_session(self)` call when a session has expired)

## Additional information
This should speed up some of our scripts that have hundreds of calls for metadata.

### PR DESCRIPTION
**Pull request by JeffreyThiessen**
## Description of changes
Removed the Hateoas links requests and the _get_link function that handles the calls. All REST request URL's are now defined in the functions.

Added (or updated) both REST error handling and IRIDA exception handling to every api request

Functionality for all calls has been kept identical to ensure there is no unintended bugs introduced in the uploaders core functionality, but TODO's have been added for future development that would improve api functionality (Likely for the graphql update)

Closes #111 

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [x] CHANGELOG.md updated with information for new change.
* [x] Tests added (or description of how to test) for any new features.
* [x] User documentation updated for UI or technical changes.

### CODE REVIEW

**Comment by JeffreyThiessen** [comment]
A build is up on https://test.pypi.org/project/iridauploader/0.8.0/ for testing

File: iridauploader/api/api_calls.py [hunk 1]
```diff
@@ -351,6 +255,9 @@ def _get_irida_exception(response):
                                                 "{status_code}: {err_msg}\n".format(
                                                     status_code=str(response.status_code),
                                                     err_msg=response.reason))
+        elif response.status_code == HTTPStatus.NOT_FOUND:
+            e = exceptions.IridaResourceError("The url you specified could not be reached. Please verify your URL and "
+                                              "the projects/samples/ect you specified.")
```
Comment (ericenns) [inline iridauploader/api/api_calls.py:260]: Very minor but should it be `etc` instead of `ect`, I am assuming that you mean etcetera.

Comment (JeffreyThiessen) [reply iridauploader/api/api_calls.py]: Good catch. Fixed.

**Review by ericenns — APPROVED** [review]
👍 looks good thanks!

**Pull request merged**